### PR TITLE
added "semi" mode to freeze

### PR DESCRIPTION
**Pull request by larryngton2**
Ported the old freeze method to a "legit" mode and added a "semi" mode which cancels PlayerMove packets sent by the client. Also made the module apply your initial velocity on disable.

### CODE REVIEW

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleFreeze.kt [hunk 1]
```diff
@@ -35,23 +35,49 @@ object ModuleFreeze : Module("Freeze", Category.MOVEMENT) {
...
+        player.pos.x = x
+        player.pos.y = y
+        player.pos.z = z
     }
 
     val packetHandler = handler<PacketEvent> { event ->
-        if (mc.world != null && event.origin == TransferOrigin.RECEIVE) {
-            if (event.packet is PlayerPositionLookS2CPacket && disableOnFlag) {
-                enabled = false
+        if (mc.world != null) {
```
Comment (ccetl) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleFreeze.kt]: World will never be `null` here, see `Module#209`.

Comment (larryngton2) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleFreeze.kt]: izuna had it there on the original freeze module

Comment (ccetl) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleFreeze.kt]: It's still useless.

Comment (1zun4) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleFreeze.kt]: Yes it is useless.

**Comment by 1zun4** [comment]
I have a feeling that this module could be a lot simpler by simply skipping the game tick handler. Are we only cancelling `PlayerMoveC2SPacket` now? This will flag any anti-cheat. Freeze was intended to act like a window right-click, where the process is being stopped for this amount of time, but without actually doing it. That usually works on any anti-cheat, as you're practically only lagging, but now you seem to only cancel the position packet, which can easily be detected as this is not possible without cheat.

What exactly are you fixing here, besides re-applying the velocity, which is an oversight.

**Comment by larryngton2** [comment]
> I have a feeling that this module could be a lot simpler by simply skipping the game tick handler. Are we only cancelling `PlayerMoveC2SPacket` now? This will flag any anti-cheat. Freeze was intended to act like a window right-click, where the process is being stopped for this amount of time, but without actually doing it. That usually works on any anti-cheat, as you're practically only lagging, but now you seem to only cancel the position packet, which can easily be detected as this is not possible without cheat.
> 
> What exactly are you fixing here, besides re-applying the velocity, which is an oversight.

both freeze methods flag as much on grim though, and the new one works better on intave. havent tested on polar

**Review by N0tHorizon — APPROVED** [review]

**Comment by larryngton2** [comment]
i have come up with an idea, ill make the freeze module have two options, "legit" which will be the old one and "semi" which will be the new one

**Comment by larryngton2** [comment]
now everyone's happy

**Comment by 1zun4** [comment]
Not me.

**Comment by EclipsesDev** [comment]
🐟

**Comment by larryngton2** [comment]
> Not me.

well what else do you want then

**Comment by 1zun4** [comment]
@larryngton2 Please test if the stationary mode works for your use-case.

**Comment by larryngton2** [comment]
> @larryngton2 Please test if the stationary mode works for your use-case.

alright

**Comment by modelpackagespaceingrootjujummail** [comment]
izuna wants to say very specific words to this guy but he is resisting the urges

**Comment by larryngton2** [comment]
> izuna wants to say very specific words to this guy but he is resisting the urges

maybe because i just started coding

**Comment by 1zun4** [comment]
PR replaced by https://github.com/CCBlueX/LiquidBounce/pull/4134

**Pull request closed without merging**